src/pieces/report.ts

```diff
@@ -16,36 +18,37 @@ async function register(bot: Client): Promise<void> {
 
 async function handleCron(bot: Client): Promise<void> {
 	const users: Array<SageUser> = await bot.mongo.collection(DB.USERS).find().toArray();
-	const courses: Array<string> = (await bot.mongo.collection(DB.COURSES).find().toArray()).map(course => course.name);
-
-	let report = `Email,Count,${courses.join(',')}\n`;
-	users.forEach(user => {
-		report += `${user.email},${user.count}`;
-		courses.forEach(course => {
-			report += `,${user.courses.includes(course)}`;
-		});
-		report += '\n';
-	});
+	const courses: Array<Course> = await bot.mongo.collection(DB.COURSES).find().toArray();
 
 	const mailer = nodemailer.createTransport({
 		host: 'mail.udel.edu',
 		port: 25
 	});
 
-	mailer.sendMail({
-		from: EMAIL.SENDER,
-		replyTo: EMAIL.REPLY_TO,
-		bcc: EMAIL.REPORT_ADDRESSES,
-		subject: 'Discord weekly Report',
-		html: `<!doctype html>
-<html>
-	<p> Here is your weekly Discord participation report. </p>
-	<p>- The <span style="color:#738ADB">Discord</span> Admin Team </p>
-</html>`,
-		attachments: [{
-			filename: `report${moment().format('M-D-YY_HH-mm-ss')}.csv`,
-			content: report
-		}]
+	const reportProfs: Array<SageUser> = users.filter(user => EMAIL.REPORT_ADDRESSES.includes(user.email));
+	reportProfs.forEach(prof => {
+		const reports: Attachment[] = [];
+		const coursesTaught = courses.filter(course => prof.roles.includes(course.roles.staff));
+		coursesTaught.forEach(course => {
+			const courseUsers = users.filter(user => user.courses.includes(course.name));
+			reports.push({
+				filename: `CISC${course.name}_${moment().format('M-D-YY_HH-mm-ss')}.csv`,
+				content: `Email,Count\n${courseUsers.map(user => `${user.email},${user.count}`).join('\n')}`
```

Comment:
```suggestion
				content: `Email, Count\n${courseUsers.map(user => `${user.email}, ${user.count}`).join('\n')}`
```
Unless the spaces are auto-added or if it just looks better without?